core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -102,6 +102,15 @@ object TopicCommand extends Logging {
     try {
       if (opts.options.has(opts.replicaAssignmentOpt)) {
         val assignment = parseReplicaAssignment(opts.options.valueOf(opts.replicaAssignmentOpt))
+        def validateReplicaAssigment: Unit = {
+          val assignmentPartition0 = assignment.getOrElse(0,
+            throw new AdminOperationException(
+              s"Unexpected replica assignment for topic '$topic', partition id 0 is missing. " +
+                s"Assignment: $assignment"))
+          val allBrokers = adminZkClient.getBrokerMetadatas()
+          adminZkClient.validateReplicaAssignment(assignment, assignmentPartition0.size, allBrokers.map(_.id).toSet)
+        }
+        validateReplicaAssigment
```

Comment:
Moving this piece of code inside *createOrUpdateTopicPartitionAssignmentPathInZK* is leading to ~30 test fails. Fixing them. Hope that should be fine.

Reply:
@omkreddy Handling the code in *createOrUpdateTopicPartitionAssignmentPathInZK* is little tricky and will lead to serious issues:
* Topic Creation - Works fine.
* Add Partitions - Fails. Let's say a topic is created with 1 partition(replicas being 0,1) and let's say broker 0 goes down after creation. Now if I have to add partitions, *alterTopic()* internally calls *createOrUpdateTopicPartitionAssignmentPathInZK* and since the existing partition 0  doesn't have all the brokers up, it fails the add partitions functionality. So it's better we handle it outside the *createOrUpdateTopicPartitionAssignmentPathInZK*.
* Topic Deletion - Works fine.

Reply:
@sumannewton  may be we can call  validateReplicaAssigment, only if update flag is false here: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/zk/AdminZkClient.scala#L96

We should try to avoid code duplication between AdminManager, TopicCommand

Reply:
Agreed @omkreddy . Even I had the same solution. Also replacing 
```
if (!replicas.toSet.subsetOf(availableBrokerIds))
        throw new BrokerNotAvailableException(
          s"Some brokers specified for partition id $partitionId are not available. " +
            s"Specified brokers: ${replicas.mkString(", ")}, " +
            s"available brokers: ${availableBrokerIds.mkString(", ")}.")
```
with
```
if ((replicas.toSet -- availableBrokerIds).size == replicas.size)
        throw new BrokerNotAvailableException(
          s"All brokers specified for partition id $partitionId are not available. " +
            s"Specified brokers: ${replicas.mkString(", ")}, " +
            s"available brokers: ${availableBrokerIds.mkString(", ")}.")
```
makes sense.
Throw error only if any partition has all unavailable replicas. With one available replica also, topic creation should happen. When those unavailable brokers come up, then they act as the replica.